journal/journal_unix.go

```diff
@@ -69,6 +69,67 @@ func Enabled() bool {
 	return true
 }
 
+// StderrIsJournalStream returns whether the process stderr is connected
+// to the Journal's stream transport.
+//
+// This can be used for automatic protocol upgrading described in [Journal Native Protocol].
+//
+// Returns true if JOURNAL_STREAM environment variable is present,
+// and stderr's device and inode numbers match it.
+//
+// Error is returned if unexpected error occurs: e.g. if JOURNAL_STREAM environment variable
+// is present, but malformed, fstat syscall fails, etc.
+//
+// [Journal Native Protocol]: https://systemd.io/JOURNAL_NATIVE_PROTOCOL/#automatic-protocol-upgrading
+func StderrIsJournalStream() (bool, error) {
+	return fdIsJournalStream(syscall.Stderr)
+}
+
+// StdoutIsJournalStream returns whether the process stdout is connected
+// to the Journal's stream transport.
+//
+// Returns true if JOURNAL_STREAM environment variable is present,
+// and stderr's device and inode numbers match it.
+//
+// Error is returned if unexpected error occurs: e.g. if JOURNAL_STREAM environment variable
+// is present, but malformed, fstat syscall fails, etc.
+//
+// This might report false negatives in certain configurations,
+// like when both stdout and stderr are independently connected to the
+// journal (the default is StandardError=inherit, which results in
+// both file descriptors getting the same duplicated socket, and doesn't
+// suffer from this problem):
+//
+//	StandardOutput=journal
+//	StandardError=journal
```

Comment:
Interesting, I wasn't aware of this behavior. If you have a small reproducer for this, I have the feeling we could open a ticket for `systemd` and directly fix this annoyance there.

Reply:
Ack, thanks for the interesting feedback anyway.
I think the behavior with `journal+console` is not that much surprising, and also an uncommon case. To avoid adding cognitive load, I'd suggest dropping the warning note.

Reply:
Done